[shopsys] remove old IE support

TomasGottvald opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| Removed old IE special css hacks, unifyed using flex width, old IE support removed from gruntile - generating special css ie file, updated autopreffixer settings for last browser 3 versions (removes red sign with using browserlist option name too).  
|New feature| No <!-- Do not forget to update docs/ -->
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ... <!-- Write "closes #123" for the issue to be closed automatically during merge -->
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

Result before changes:
![grunt_before_changes](https://user-images.githubusercontent.com/35454496/66560501-d2a19e00-eb57-11e9-891c-12f5fa308167.png)
CSS file size: 84kb + 95kb for old IE

![grunt_after_change](https://user-images.githubusercontent.com/35454496/66560541-e51bd780-eb57-11e9-8ea9-d762eeab1ddf.png)
CSS file size: 66.8kb (no other css files)

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/chooser.less:
```diff
@@ -17,55 +17,30 @@
 
         &__title {
-            flex: 1;
+            width: @box-chooser-title-width;
```
Is this correct? I think `flex: 1;` does not equal with `width: @box-chooser-title-width;`. Please check it.

TomasGottvald replied:
Fixed adding flex: 1 back. thx

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width.

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/error-page.less:
```diff
@@ -3,25 +3,11 @@
-                float: left;
-                width: 100%/2;
-            }
+            width: 100%/2;
```
This is the same case as the comment above. Please check if this is a correct solution.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width which will be same for both sides 50%

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/error-page.less:
```diff
@@ -30,20 +16,11 @@
-                float: left;
-                width: 100%/2;
-            }
+            width: 100%/2;
```
Please check if this is a correct solution.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/box/progress.less:
```diff
@@ -7,25 +7,14 @@
-
     &__item {
-        flex: 1;
+        width: 100%/@box-progress-steps-count;
```
Please check if you can substitute `flex` property for the `width` property.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width.

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/action.less:
```diff
@@ -7,22 +7,12 @@
         @media @query-md {
+            width: 100%/2;
             order: 1;
-            flex: 1;
```
Same as the comment above.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width same for both 50%.

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/action.less:
```diff
@@ -31,14 +21,9 @@
         @media @query-md {
             order: 2;
-            flex: 1;
+            width: 100%/2;
```
Same as the comment above.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width same for both 50%.

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/categories.less:
```diff
@@ -40,40 +40,22 @@
 
         &__title {
-            flex: 1;
+            width: calc(~"100% - @{list-categories-item-image-width}");
```
Please check if you can substitute `flex` property for the `width` property.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/products-line.less:
```diff
@@ -28,73 +28,42 @@
 
         &__title {
-            flex: 1;
+            width: @list-products-line-item-title-width-mobile;
```
Please check if you can substitute `flex` property for the `width` property.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/list/products-line.less:
```diff
@@ -28,73 +28,42 @@
-                    width: @list-products-line-item-title-width;
-                }
+            @media @query-md {
+                width: @list-products-line-item-title-width;
```
Please check if you can substitute `flex` property for the `width` property.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/window/popup.less:
```diff
@@ -27,12 +27,12 @@
     border-radius: @radius;
 
-    .is-no-flex &, .is-safari &, .is-flex-popup-height-issue-detected & {
+    .is-safari &, .is-flex-popup-height-issue-detected & {
```
Is the class `is-safari` still necessary?

TomasGottvald replied:
really dont know ... but will be checked later during refactoring

topolanekmartin replied:
Okay, let's refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/header/search.less:
```diff
@@ -7,35 +7,21 @@
-
         &__input {
-            flex: 1;
+            width: ~"calc(100% - @{search-button-width})";
```
Please check if you can substitute `flex` property for the `width` property.

TomasGottvald replied:
Flex: 1 sets all flexible items to be same lenght. Is not necessary bcs, we set calculated width

topolanekmartin replied:
It is okay for now. We will refactor this later.

topolanekmartin commented on project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/layout/layout.less:
```diff
@@ -12,7 +12,7 @@ img {
     min-height: 100%;
 
-    .is-no-flex &, .is-safari & {
+    .is-safari & {
```
Is the class `is-safari` still necessary?

TomasGottvald replied:
really dont know ... but will be checked later during refactoring

topolanekmartin replied:
Okay, let's refactor this later.

TomasGottvald commented:
> 
> 
> Hi, thanks for this PR. The code looks okay to me but there two things that I'm missing here.
> 
>     1. please add upgrade notes
> 
>     2. did you check administration as well? If so, why didn't you remove the support there as well?
1) Added upgrade notes
2) Good point. I checked admin too. All these old hack were deleted during admin facelift. I just unified flex width definition and removed old ie metatags.

DavidOstrozlik commented on upgrade/UPGRADE-v8.1.0-dev.md:
````diff
@@ -372,6 +372,66 @@ There you can find links to upgrade notes for other versions too.
+
+        - grunt.registerTask('adminLess', ['less:admin', 'legacssy:admin' ]);
+        + grunt.registerTask('adminLess', ['less:admin']);
+        ```
````
```suggestion
        ```
        - for more information you can see the [PR](https://github.com/shopsys/shopsys/pull/1461)
```
